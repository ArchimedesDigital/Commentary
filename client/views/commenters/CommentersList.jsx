CommentersList = React.createClass({

  mixins: [ReactMeteorData],

  propTypes: {
    limit: React.PropTypes.number,
		featureOnHomepage: React.PropTypes.bool,
  },

  getMeteorData() {
    let query = {};

    var limit = 100;

    if(this.props.limit){
      limit = this.props.limit;
    }
		if(this.props.featureOnHomepage){
			query.featureOnHomepage = this.props.featureOnHomepage;
		}

    return {
      commenters: Commenters.find(query, {sort: {name: 1}, limit: limit}).fetch(),
    };
  },

  renderCommenters() {

    return this.data.commenters.map((commenter) => {
      return <CommenterTeaser
              key={commenter._id}
              commenter={commenter} />;

    });

  },

  render() {

     return (
       <div className="commenters-list">

         {this.renderCommenters()}

       </div>


      );
    }


});
