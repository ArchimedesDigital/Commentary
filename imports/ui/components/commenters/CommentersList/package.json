{
  "name": "CommentersList",
  "version": "0.0.1",
  "private": true,
  "main": "./CommentersList.js"
}
