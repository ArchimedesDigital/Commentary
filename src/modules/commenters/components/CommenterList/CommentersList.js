import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { compose } from 'react-apollo';
import commentersQuery from '../../graphql/queries/commentersQuery';


// components
import CommenterTeaser from '../CommenterTeaser/CommenterTeaser';


class CommentersList extends Component { 
	constructor(props) {
		super(props);
		this.state = {};
	}
	componentWillReceiveProps(nextProps) {
		let commenters = [];
		const tenantId = sessionStorage.getItem('tenantId');
		const properites = {
			tenantId: tenantId
		};
		nextProps.commentersQuery.refetch(properites);
		commenters = nextProps.commentersQuery.loading ? [] : nextProps.commentersQuery.commenters;
		// SUBSCRIPTIONS:
		if (nextProps.featureOnHomepage) {
			commenters = nextProps.commentersQuery.loading ? [] : nextProps.commentersQuery.commenters
			.filter(x => x.featureOnHomepage === true);
		}
		this.setState({
			commenters: commenters
		});
	}
	render() {
		const { commenters } = this.state;
		if (!commenters) {
			return null;
		}
		return (
			<div className="commenters-list">
				{commenters.map(commenter =>
					(<CommenterTeaser
						key={commenter._id}
						commenter={commenter}
					/>)
				)}
			</div>
		);
	}
}
CommentersList.propTypes = {
	commentersQuery: PropTypes.object,
	featureOnHomepage: PropTypes.bool,
	limit: PropTypes.number
};
CommentersList.defaultProps = {
	commenters: [],
};

export default compose(commentersQuery)(CommentersList);
