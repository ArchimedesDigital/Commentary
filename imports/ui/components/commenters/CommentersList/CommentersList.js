import React from 'react';
import PropTypes from 'prop-types';
import { Meteor } from 'meteor/meteor';
import { Session } from 'meteor/session';
import { createContainer } from 'meteor/react-meteor-data';
import { compose } from 'react-apollo';
import { commentersQuery } from '/imports/graphql/methods/commenters';

// models
import Commenters from '/imports/models/commenters';

// components
import CommenterTeaser from '/imports/ui/components/commenters/CommenterTeaser';


const CommentersList = ({ commenters }) => (
	<div className="commenters-list">
		{commenters.map(commenter =>
			(<CommenterTeaser
				key={commenter._id}
				commenter={commenter}
			/>)
		)}
	</div>
);
CommentersList.propTypes = {
	commenters: PropTypes.arrayOf(PropTypes.shape({
		_id: PropTypes.string.isRequired,
	}).isRequired),
};
CommentersList.defaultProps = {
	commenters: [],
};


const cont = createContainer(({ limit, featureOnHomepage }) => {
	let commenters = [];
	let _limit = 100;
	if (limit) {
		_limit = limit;
	}

	// SUBSCRIPTIONS:
	if (featureOnHomepage) {
		Meteor.subscribe('commenters.featureOnHomepage', Session.get('tenantId'), _limit);
		commenters = Commenters.find({
			featureOnHomepage: true,
		}, {
			sort: {
				name: 1,
			},
			_limit,
		}).fetch();
	} else {
		Meteor.subscribe('commenters', Session.get('tenantId'), _limit);
		commenters = Commenters.find({}, { sort: { name: 1 }, _limit }).fetch();
	}

	return {
		commenters,
	};
}, CommentersList);
export default (commentersQuery)(cont);
