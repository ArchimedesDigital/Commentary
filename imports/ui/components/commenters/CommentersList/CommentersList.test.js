import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentersList from './CommentersList';

describe('CommentersList', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentersList />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
