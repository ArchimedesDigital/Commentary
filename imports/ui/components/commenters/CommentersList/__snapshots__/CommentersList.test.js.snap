// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentersList renders correctly 1`] = `
<div
  className="commenters-list"
/>
`;
